samples/create_cluster.js

```diff
@@ -0,0 +1,170 @@
+    const [longRunningOp] = await client.getOperation({name: opId});
+    const DONE = STATUS_ENUM[STATUS_ENUM.DONE];
+    if (longRunningOp.status === DONE) {
+      resolve('Cluster creation completed.');
+    } else {
+      reject('Cluster creation not complete.');
+    }
+  };
+  return new Promise(getOpFn);
+};
+
+/**
+ * A simple function that returns the next number in a fibonacci sequence that
+ * has the given argument as f(n-1) and the global variable `prevFibonacciDelay`
+ * as f(n - 2). This function is used to simulate a fibonacci sequence for the
+ * backing off (delay) time used when retrying.
+ *
+ * @param {number} delay the previous delay in the sequence
+ * @returns the next delay before retrying
+ */
+function getFibonacciDelay(delay) {
+  const newDelay = prevFibonacciDelay + delay;
+  prevFibonacciDelay = delay;
+  return newDelay;
+}
+
+/**
+ * This method calls the Promise function (`checkOpStatus()`) to check the
+ * status of an operation and schedules a retry if the operation is still not
+ * completed. The retry delay is modelled as a fibonacci sequence.
+ *
+ * @param {container.v1.ClusterManagerClient} client the google cloud API client used to submit the request
+ * @param {string} opId the unique identifier of the operation we want to check
+ * @param {number} delay the delay with which the next retry is to be scheduled
+ * @param {number} maxtries maximum number of retry attempts to go before giving up
+ * @param {number} retryCount the current number of retried attempted
+ */
+function checkStatusWithRetry(client, opId, delay, maxtries, retryCount = 1) {
+  checkOpStatus(client, opId)
+    .then(status => {
+      // success scenario
+      console.log(status);
+    })
+    .catch(status => {
+      // fail (not yet complete) scenario
+      if (retryCount < maxtries) {
+        console.log(`${status} will try after ${delay / 1000}s delay...`);
+        const newDelay = getFibonacciDelay(delay);
+        setTimeout(
+          () =>
+            checkStatusWithRetry(
+              client,
+              opId,
+              newDelay,
+              maxtries,
+              retryCount + 1
+            ),
+          delay
+        );
+      } else {
+        console.log(`${status} max retries reached, giving up.`);
+      }
+    });
+}
+
+/**
+ * This method creates a GKE cluster using the given client in the GCP project
+ * and zone indicated by the `gcpProject` and `gcpZone` parameters. The last
+ * argment is used as the name given to the newly created cluster.
+ *
+ * @param {container.v1.ClusterManagerClient} client the google cloud API client used to submit the request
+ * @param {string} gcpProject the GCP project where the cluster is to be created
+ * @param {string} gcpZone the GCP zone where the cluster is to be created
+ * @param {string} clusterName the name to be given to the new cluster
+ */
+async function createCluster(gcpZone, clusterName, networkName) {
```

Comment:
I think this should actually go inside main, and then called in main as well (see guidelines).

Reply:
Fixed in https://github.com/googleapis/nodejs-cloud-container/pull/519/commits/ee670d950b4f5a8e53ec494e4448b2bd2ca38e6b